04_設計/04_プラハチャレンジをDDDで実装してみる/02_APIサーバ/src/presentation/controller/member/get-member-controller.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  IController,
+  MiddlewareServices,
+  RequestServices,
+  ResponseServices,
+} from "presentation/interface/controller";
+import { MemberRouteRegistrar } from "presentation/route-registrar";
+import * as Usecase from "usecase";
+
+export class GetMemberController implements IController {
+  public constructor(
+    private readonly getAllMemberUsecase: Usecase.GetAllMember,
+    private readonly getMemberByTeamNameUsecase: Usecase.GetMemberByTeamName,
+    private readonly getMemberByPairNameUsecase: Usecase.GetMemberByPairName,
+    private readonly getMemberByExerciseAndProgressUsecase: Usecase.GetMemberByExerciseAndProgress,
+  ) {}
+
+  public execute = async (
+    requestServices: RequestServices,
+    responseServices: ResponseServices,
+    middleWareServices: MiddlewareServices,
+  ): Promise<void> => {
+    const { getQueryParams } = requestServices;
+    const { setResponseStatus, sendResponse } = responseServices;
+    const { nextFunction } = middleWareServices;
+
+    try {
+      const params = getQueryParams.execute();
+
+      this.validateParams(params);
+
+      if (this.isGetAllMemberUsecaseParamsPattern(params)) {
```

Comment:
[IMO]
以下のように呼び出す形に作っておくほうが好きです！
こうすれば、paramsの定義にチェック処理を実装できるので！
```suggestion
      if (params.isGetAllMemberUsecaseParamsPattern()) {
```

Reply:
ここちょっと実装イメージがわからないです:eyes:
ここのparamsはgetQueryParams.execute()の返り値で、型は`{ [key: string]: string }`です。
paramsにパラメータのパターンを判定するロジックは実装できないような気がしているのですが、、